### PR TITLE
see issue #212

### PR DESCRIPTION
**Pull request by esohns-humana**
switch from OpenLayers.Markers to OpenLayers.Feature.Vector …
- OpenLayers.Marker is obsolete
- allows easy integration of drag&drop support for Markers

improves ovi support
- missing click event delegation
- allow reverse geocoding

### CODE REVIEW

**Comment by vicchi** [comment]
The definition of `Marker.toProprietary` in `mxn.openlayers.core.js` seems to contain a redundant definition of `var style` and `var marker` around line 670 in your version of this file. Which set of definitions should be used ... the first or the second?

File: source/mxn.openlayers.core.js [hunk 1]
```diff
@@ -530,85 +645,72 @@ mxn.register('openlayers', {
...
+			 graphicXOffset : anchor.x,
+			 graphicYOffset : anchor.y,
+			 graphicZIndex  : (!!this.attributes.zIndex ? this.attributes.zIndex : 2)//,
+			 // title          : this.labelText
+		    };
+			var marker = new OpenLayers.Feature.Vector(new OpenLayers.Geometry.Point(this.location.toProprietary('openlayers').lon,
+			                                                                         this.location.toProprietary('openlayers').lat),
+													   null,
+													   style);
+			var style = {
```
Comment (vicchi) [inline source/mxn.openlayers.core.js]: The second (redundant?) definition of `marker` and `style` is here. Is this the authoritative one, or the previous?

Comment (esohns-humana) [inline source/mxn.openlayers.core.js]: Should be fixed now, sorry for the confusion.

Comment (vicchi) [inline source/mxn.openlayers.core.js]: Great, got the updated version now. Thanks

**Comment by vicchi** [comment]
Manually merged into `release-2.1` branch; see commits e80ff3414c6d2d4baea396255604a1637bc8b940, 1f041e106fd74ae562f02de6dd51d9e484592c51, 692a266e9a2d7a3a22e715d05b8b0d5086b22246 and 0c778da0b7e680a0d3c1e7e3c623a4f310c4525c.

**Comment by vicchi** [comment]
I'm re-opening this Pull Request as there's something not quite right going on. The Pull Request says _switch from OpenLayers.Markers to OpenLayers.Feature.Vector_ (because) _OpenLayers.Marker is obsolete_.

As a result, the MXN equivalent of a Marker in OpenLayers now no longer displays a push-pin style icon.

Furthermore, I can't find any reference in the release notes for OpenLayers 2.12 that `OpenLayers.Marker` is obsolete and this class is still documented in the API documentation at http://dev.openlayers.org/releases/OpenLayers-2.12/doc/apidocs/files/OpenLayers/Marker-js.html.

@esohns-humana - can you help out here please? At the moment, I'm minded to roll back the changes for this Pull Request as it's breaking core MXN functionality.

**Comment by vicchi** [comment]
In the interim, I've created a new branch, `release-2.1-openlayers` and rolled back to commit 78aab568bef8eadc2e2be2dfd257b713374d7d9b. OpenLayers marker icons now work again and the core tests are also working.

**Comment by esohns-humana** [comment]
Well, it does display a push-pin style icon... if you set one manually. The default case was broken.
Please re-pull.

With respect to the obsoletion of <code>Openlayers.Marker</code>: there are many references to this in the forum. Hint: search for information on how to drag a marker... Feel free to correct me on this.

**Comment by vicchi** [comment]
I'll re-pull. Thanks. Have you got a forums link? I looked but couldn't see anything on obsoletion of `OpenLayers.Marker` other than recommending use of `OpenLayers.Control.DragMarker` for draggable markers although admittedly I made a cursory search.

**Comment by esohns-humana** [comment]
FYI: <link>http://comments.gmane.org/gmane.comp.gis.openlayers.user/6031</link>

**Comment by vicchi** [comment]
Ah. I see. So if Christopher Schmidt is advocating a move to vectors for markers then even if it's not obsoleted in the current version there's a high chance it will be, maybe in the next version, maybe in version 3.0.

I'll take a look at the updated pull request and providing there's no regression of core functionality and it passes the build tests and core tests I'll merge it in.

**Comment by esohns-humana** [comment]
Agreed. As he is the author of <code>OpenLayers.Control.DragMarker</code>, I propose to follow his take on this matter.

**Comment by vicchi** [comment]
Agreed again. If the author of `OpenLayers.Control.DragMarker` is advocating that this never makes it into a formal release and that `OpenLayers.Control.DragFeature` is used instead that's good enough for me.

I've merged your latest pull request into my local build and all looks good in the core tests. But there's some `jslint` errors being thrown in the build so I'll fix those and then I'll commit and push to the `release-2.1` branch later today or tomorrow.

**Comment by vicchi** [comment]
This is now merged back into the `release-2.1` branch; see commit 5d945f61e686c77b4758a2b5ee89c2395536af39

**Pull request closed without merging**